File: bindings/python/examples/srcml_create_archive_filename.py
#!/usr/bin/python
#
# @file srcml_create_archive_filename.py
#
# @copyright Copyright (C) 2013-2014 srcML, LLC. (www.srcML.org)
#
# The srcML Toolkit is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# The srcML Toolkit is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with the srcML Toolkit; if not, write to the Free Software
#Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 


# Example program of the use of the C API for srcML.

# Create an archive, file by file, with an output filename.


import sys, srcml

if __name__ == "__main__":
    archive = None
    unit = None

    # create a new srcml archive structure
    archive = srcml.create_archive()

    # open a srcML archive for output
    srcml.write_open_filename(archive, "srcml_create_archive_filename.xml")

    # add all the files to the archive
    for arg in sys.argv[1:]:
        unit = srcml.create_unit(archive)
        srcml.unit_set_filename(unit, arg)

        # Translate to srcml and append to the archive 
        srcml.parse_unit_filename(unit, arg)

        # Translate to srcml and append to the archive
        srcml.write_unit(archive, unit)

        srcml.free_unit(unit)

    # close the srcML archive
    srcml.close_archive(archive)

    # free the srcML archive data
    srcml.free_archive(archive)

File: examples/libsrcml/srcml_create_archive_filename.cpp
// SPDX-License-Identifier: GPL-3.0-only
/**
 * @file srcml_archive_create_filename.cpp
 *
 * @copyright Copyright (C) 2013-2019 srcML, LLC. (www.srcML.org)
 *
 * Example program of the use of the libsrcml C API.
 *
 * Create an archive, file by file, with an output filename.
*/

#include <srcml.h>

int main(int argc, char* argv[]) {

    // create a new srcml archive structure
    srcml_archive* archive = srcml_archive_create();

    // open a srcML archive for output
    srcml_archive_write_open_filename(archive, "newProject.xml");

    // add all the files to the archive
    for (int i = 1; i < argc; ++i) {

        srcml_unit* unit = srcml_unit_create(archive);

        // set the filename
        srcml_unit_set_filename(unit, argv[i]);

        // translate to srcml
        srcml_unit_parse_filename(unit, argv[i]);

        // append to the archive
        srcml_archive_write_unit(archive, unit);

        srcml_unit_free(unit);
    }

    // close the srcML archive
    srcml_archive_close(archive);

    // free the archives
    srcml_archive_free(archive);

    return 0;
}
